components/brave_wallet/browser/bitcoin/bitcoin_rpc.cc

```diff
@@ -0,0 +1,251 @@
+  if (!IsAsciiAlphaNumeric(address) || !IsAsciiAlphaNumeric(last_seen_txid)) {
+    return GURL();
+  }
+
+  GURL::Replacements replacements;
+  std::string path = base::StrCat(
+      {base_url.path(),
+       base::JoinString({"address", address, "txs", "chain"}, "/")});
+  if (!last_seen_txid.empty()) {
+    base::StrAppend(&path, {"/", last_seen_txid});
+  }
+  replacements.SetPathStr(path);
+
+  return base_url.ReplaceComponents(replacements);
+}
+
+const GURL MakePostTransactionUrl(const GURL& base_url) {
+  GURL::Replacements replacements;
+  const std::string path = base::StrCat({base_url.path(), "tx"});
+  replacements.SetPathStr(path);
+
+  return base_url.ReplaceComponents(replacements);
+}
+
+std::string BitcoinRpcHost() {
+  return base::CommandLine::ForCurrentProcess()->GetSwitchValueASCII(
+      brave_wallet::switches::kBitcoinRpcHost);
+}
+
+GURL BaseRpcUrl(const std::string& network_id) {
+  CHECK(brave_wallet::IsBitcoinNetwork(network_id));
+
+  std::string host = BitcoinRpcHost();
+  const char* path = network_id == brave_wallet::mojom::kBitcoinMainnet
+                         ? "/api/"
+                         : "/testnet/api/";
+
+  return GURL(base::StrCat(
+      {url::kHttpsScheme, url::kStandardSchemeSeparator, host, path}));
+}
+
+absl::optional<std::string> ConvertPlainIntToJsonArray(
+    const std::string& json) {
+  return base::StrCat({"[", json, "]"});
+}
+
+absl::optional<std::string> ConvertPlainStringToJsonArray(
+    const std::string& json) {
+  return base::StrCat({"[\"", json, "\"]"});
+}
+
+}  // namespace
+
+namespace brave_wallet {
+
+BitcoinRpc::BitcoinRpc(
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : api_request_helper_(new APIRequestHelper(GetNetworkTrafficAnnotationTag(),
+                                               url_loader_factory)) {}
+
+BitcoinRpc::~BitcoinRpc() = default;
+
+void BitcoinRpc::GetChainHeight(const std::string& network_id,
+                                GetChainHeightCallback callback) {
+  auto internal_callback =
+      base::BindOnce(&BitcoinRpc::OnGetChainHeight,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+  // Response comes as a plain integer which is not accepted by json sanitizer.
+  // Wrap response into a json array.
+  auto conversion_callback = base::BindOnce(&ConvertPlainIntToJsonArray);
+  RequestInternal(true, MakeGetChainHeightUrl(BaseRpcUrl(network_id)),
+                  std::move(internal_callback), std::move(conversion_callback));
+}
+
+void BitcoinRpc::OnGetChainHeight(GetChainHeightCallback callback,
+                                  APIRequestResult api_request_result) {
+  if (!api_request_result.Is2XXResponseCode()) {
+    std::move(callback).Run(base::unexpected("Unexpected HTTP result code"));
+    return;
+  }
+
+  auto* list = api_request_result.value_body().GetIfList();
+  if (!list) {
+    std::move(callback).Run(base::unexpected("Json response is not an arary"));
```

Comment:
sed -i 's/arary/array/g' components/brave_wallet/browser/bitcoin/bitcoin_rpc.cc

Reply:
Oops:)
I'll fix it in this PR if anything else is requested or in the next one.